.github/workflows/common-java-gradle-build.yml

```diff
@@ -0,0 +1,45 @@
+# This is a basic build workflow for Octopus gradle Java modules
+name: Gradle Compile & UT
+
+on:
+  workflow_call:
+    inputs:
+      java-version:
+        required: false
+        type: string
+        default: '8'
+      docker-image:
+        required: false
+        type: string
+        default: ''
+      build-version:
+        required: false
+        type: string
+        default: '1.0-SNAPSHOT'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    environment: Prod
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ inputs.java-version }}
+          distribution: 'adopt'
+          cache: 'gradle'
+
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Gradle build
+        if: ${{ inputs.docker-image == '' }}
+        run: ./gradlew build -P buildVersion=${{ inputs.build-version }}
+
+      - name: Gradle build docker image
```

Comment:
If I have understood correctly the assumption is that `inputs.docker-image` should be defined only in case of hybrid-flow applications (we skip tests in that case).

Therefore we will not be able to use "GH-only flow" for application publishing some docker image. Is it correct?